external/buildscripts/build.pl

```diff
@@ -1159,20 +1154,37 @@
 			$existingMonoRootPath = "$monoInstalls/$monoVersionToUse";
 		}
 
+		if ($targetArch eq "arm64")
+		{
+			$macversion = "11.0"; # To build on ARM64, we need to specify min OS version as 11.0 as we need to use new APIs from 11.0
+		}
+
 		$mcs = "EXTERNAL_MCS=$existingMonoRootPath/bin/mcs";
 
 		$ENV{'CC'} = "$macSdkPath/../usr/bin/clang";
 		$ENV{'CXX'} = "$macSdkPath/../usr/bin/clang++";
-		$ENV{'CFLAGS'} = $ENV{MACSDKOPTIONS} = "-D_XOPEN_SOURCE -I$macBuildEnvDir/builds/usr/include -mmacosx-version-min=$macversion -isysroot $macSdkPath";
+		$ENV{'CFLAGS'} = $ENV{MACSDKOPTIONS} = "-mmacosx-version-min=$macversion -isysroot $macSdkPath -g";
```

Comment:
Not sure if we always want debug symbols (`-g`)

Reply:
I think we do. I also made the change in Unity build code to strip them out for non-development builds: https://ono.unity3d.com/unity/unity/pull-request/108590/_/platform/mac/arm64/buildcode

Without this change, debugging mono source code in Unity isn't possible unless you rebuild it as debug, which is very burdensome.

This change makes macOS match Windows (where we always produce PDBs).